app/colorCoder.js

```diff
@@ -1,99 +1,36 @@
-            }
-
-        return (majorIndex * MinorColorNames.length) + (minorIndex + 1);
-	 }
-	 
-	 function test(){
-			pairNumber = 4;
-            let testPair1 = getColorFromPairNumber(pairNumber);
-            console.log(`[In]Pair Number: ${pairNumber},[Out] Colors:${testPair1}`);
-			console.assert(testPair1.majorColor == "WHITE");
-			console.assert(testPair1.minorColor == "BROWN");
-
-			pairNumber = 5;
-            testPair1 = getColorFromPairNumber(pairNumber);
-            console.log(`[In]Pair Number: ${pairNumber},[Out] Colors:${testPair1}`);
-			console.assert(testPair1.majorColor == "WHITE");
-			console.assert(testPair1.minorColor == "SLATEGRAY");
-			 
-			pairNumber = 23;
-            testPair1 = getColorFromPairNumber(pairNumber);
-            console.log(`[In]Pair Number: ${pairNumber},[Out] Colors:${testPair1}`);
-			console.assert(testPair1.majorColor == "RED");
-			console.assert(testPair1.minorColor == "GREEN");
-			 
-            let testPair2 = new ColorPair();
-			testPair2.majorColor="YELLOW";
-			testPair2. minorColor ="GREEN";
-            pairNumber =getPairNumberFromColor(testPair2);
-            console.log(`[In]Colors: ${testPair2}, [Out] PairNumber: ${pairNumber}`);
-			console.assert(pairNumber==18);
-
-			testPair2 = new ColorPair();
-			testPair2.majorColor="RED";
-			testPair2. minorColor ="BLUE";
-            pairNumber =getPairNumberFromColor(testPair2);
-            console.log(`[In]Colors: ${testPair2}, [Out] PairNumber: ${pairNumber}`);
-			console.assert(pairNumber==6);
-}
-test();
+import { getPairNumberFromColor } from "./getPairNumberFromColor.js";
+import { getColorFromPairNumber } from "./getColorFromPairNumber.js";
+import { ColorPair } from "./colorPair.js";
+import { getReferenceManual } from "./referenceManual.js";
+
+function testGetColorFromPairNumber(pairNumber, majorColor, minorColor) {
+  let testPair = getColorFromPairNumber(pairNumber);
+  console.log(`[In]Pair Number: ${pairNumber}, [Out] Colors:${testPair}`);
+  console.assert(testPair.majorColor == majorColor);
+  console.assert(testPair.minorColor == minorColor);
+}
+
+function testGetPairNumberFromColor(majorColor, minorColor, testPairNumber) {
+  let testPair = new ColorPair();
+  testPair.majorColor = majorColor;
+  testPair.minorColor = minorColor;
+  let pairNumber = getPairNumberFromColor(testPair);
+  console.log(`[In]Colors: ${testPair}, [Out] PairNumber: ${pairNumber}`);
+  console.assert(pairNumber == testPairNumber);
+}
+
+function test() {
+  testGetColorFromPairNumber(4, "WHITE", "BROWN");
+  testGetColorFromPairNumber(5, "WHITE", "SLATEGRAY");
+  testGetColorFromPairNumber(23, "RED", "GREEN");
+  testGetPairNumberFromColor("YELLOW", "GREEN", 18);
+  testGetPairNumberFromColor("RED", "BLUE", 6);
+}
+
+console.log("----------------TEST CASES----------------");
+test();
+
+console.log("------------REFERENCE MANUAL--------------");
+console.table(getReferenceManual());

```

Comment:
Try asserting on the output of `getReferenceManual` - it's easy to do in your design, because you have separated the formation of the manual from its printing.
Some tests to safeguard the functionality: Assert that the numbers are 1-based and not 0-based, assert that no combination is left out. Try these asserts.